feat: add `math/base/special/asinf`

gunjjoshi opened the pull request:
Resolves #1944.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `math/base/special/asinf`, which computes the arcsine of a single precision floating point number.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   fixes #1944.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gunjjoshi committed:
feat: add math/base/special/asinf

gunjjoshi committed:
Update README.md

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update repl.txt

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update README.md

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update repl.txt

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update main.c

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update main.c

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/docs/types/index.d.ts:
```diff
@@ -0,0 +1,48 @@
+*
+* @example
+* var v = asinf( 0.0 );
+* // returns ~0.0
```
```suggestion
* // returns 0.0
```

In the README, we have this as exact. I'd expect the same here.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+*
+* @example
+* var v = asinf( 0.0 );
+* // returns ~0.0
```
```suggestion
* // returns 0.0
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+*
+* ## Method
+*
+* -   A rational function of the form
```
```suggestion
* -   A polynomial of the form
```

gunjjoshi replied:
We might need to change this [here](https://github.com/stdlib-js/stdlib/blob/55647661f551a753ba247511606eba2ee4766db8/lib/node_modules/%40stdlib/math/base/special/asin/lib/main.js#L56) too @kgryte .

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
````diff
@@ -0,0 +1,144 @@
+* -   A rational function of the form
+*
+*     ```tex
+*     x + x^3 \frac{P(x^2)}{Q(x^2)}
````
```suggestion
*     x + x^3 P(x^2)
```

kgryte replied:
I'm taking this from the Cephes source for `asinf`.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+*
+*     | arithmetic | domain | # trials | peak    | rms     |
+*     |:-----------|:-------|:---------|:--------|:--------|
+*     | IEEE       | -1, 1  | 10^5     | 1.4e-7  | 4.2e-8  |
```
```suggestion
*     | IEEE       | -1, 1  | 10^5     | 2.5e-7  | 5.0e-8  |
```

Taken from the Cephes source.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+	if ( x > 0.0 ) {
+		sgn = 1;
+		a = x;
+	}
+	else {
```
```suggestion
	} else {
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+		return 0.0;
+	}
+	if ( a < 1.0e-4 ) {
+		z = float64ToFloat32( a );
+		if ( sgn < 0 ) {
+			z = -z;
+		}
+		return z;
```
```suggestion
		if ( sgn < 0 ) {
			a = -a;
		}
		return a;
```

This reassignment is not necessary based on how you've implemented this based on the source implementation.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+		z = float64ToFloat32( 0.5 * float64ToFloat32( 1.0 - a ) );
+		x = sqrtf( z );
+		flag = 1;
+	}
+	else {
```
```suggestion
	} else {
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+		flag = 1;
+	}
+	else {
+		x = float64ToFloat32( a );
```
```suggestion
		x = a;
```

You've already converted to float32 at L99.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+		flag = 0;
+	}
+	// eslint-disable-next-line max-len
+	z = float64ToFloat32( float64ToFloat32( ( float64ToFloat32( polyp( z ) ) * float64ToFloat32( z * x ) ) ) + ( x ) );
```
```suggestion
	z = float64ToFloat32( float64ToFloat32( polyp( z ) * float64ToFloat32( z * x ) ) + x );
```

A few too many parentheses and conversions.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,144 @@
+	z = float64ToFloat32( float64ToFloat32( ( float64ToFloat32( polyp( z ) ) * float64ToFloat32( z * x ) ) ) + ( x ) );
+
+	if ( flag !== 0 ) {
+		z += z;
```
```suggestion
		z = float64ToFloat32( z + z );
```

Arithmetic operation so we need to wrap in order to emulate float32 arithmetic.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -0,0 +1,141 @@
+		z = float64ToFloat32( x * x );
+		flag = 0;
+	}
+	// eslint-disable-next-line max-len
+	z = float64ToFloat32( float64ToFloat32( polyp( z ) * float64ToFloat32( z * x ) ) + x );
```
```suggestion
	z = float64ToFloat32( float64ToFloat32( polyp( z ) * float64ToFloat32( z * x ) ) + x ); // eslint-disable-line max-len
```

We typically prefer in-line ESLint comments.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/native.js:
```diff
@@ -0,0 +1,58 @@
+*
+* @example
+* var v = asinf( 0.0 );
+* // returns ~0.0
```
```suggestion
* // returns 0.0
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/package.json:
```diff
@@ -0,0 +1,70 @@
+  "directories": {
+    "benchmark": "./benchmark",
+    "doc": "./docs",
+    "example": "./examples",
+    "lib": "./lib",
+    "scripts": "./scripts",
```
```suggestion
    "example": "./examples",
    "include": "./include",
    "lib": "./lib",
    "scripts": "./scripts",
    "src": "./src",
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/package.json:
```diff
@@ -0,0 +1,72 @@
+    "trig",
+    "trigonometry",
+    "radians",
+    "angle",
+    "polyfill",
+    "ponyfill"
```
```suggestion
    "angle"
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/scripts/evalpoly.js:
```diff
@@ -0,0 +1,128 @@
+
+// Polynomial coefficients ordered in ascending degree...
+
+/*
+* arcsin(x) = x + x^3 P(x^2)/Q(x^2)
+* 0 <= x <= 0.625
+* Peak relative error = 1.2e-18
+*/
```
```suggestion
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/scripts/evalpoly.js:
```diff
@@ -0,0 +1,123 @@
+// VARIABLES //
+
+// Polynomial coefficients ordered in ascending degree...
+
```
```suggestion
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+* @return     evaluated polynomial
+*/
+static float poly_p( const float x ) {
+	return 0.16666752422 + (x * (0.074953002686f + (x * (0.045470025998f + (x * (0.024181311049f + (x * 0.042163199048f)))))));
```
Looks like there's a bug in `evalpoly-compile-c`. The first value is missing its suffix.

gunjjoshi replied:
Yes @kgryte . I looked into it. I believe, we need to add :

```C
else {
	horner += '{{dtype_suffix}}';
}
```

after line 124 at https://github.com/stdlib-js/stdlib/blob/55647661f551a753ba247511606eba2ee4766db8/lib/node_modules/%40stdlib/math/base/tools/evalpoly-compile-c/lib/main.js#L124

kgryte replied:
Let me push a fix now.

kgryte replied:
Pushed. The latest `develop` should have the patch which you can then rerun to get the correct polynomial function.

kgryte replied:
@gunjjoshi You'll want to rerun the evalpoly script.

gunjjoshi replied:
Oh, that went out of my mind. Will do it now.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+*
+* @example
+* float out = stdlib_base_asinf( 0.0f );
+* // returns ~0.0f
```
```suggestion
* // returns 0.0f
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+*     | IEEE       | -1, 1  | 10^5     | 1.4e-7  | 4.2e-8  |
+*
+* @param x    input value
+* @returns    arcsine (in radians)
```
```suggestion
* @return    arcsine (in radians)
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+*
+* ## Method
+*
+* -   A rational function of the form
```
```suggestion
* -   A polynomial of the form
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
````diff
@@ -0,0 +1,137 @@
+* -   A rational function of the form
+*
+*     ```tex
+*     x + x^3 \frac{P(x^2)}{Q(x^2)}
````
```suggestion
*     x + x^3 P(x^2)
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+*
+*     | arithmetic | domain | # trials | peak    | rms     |
+*     |:-----------|:-------|:---------|:--------|:--------|
+*     | IEEE       | -1, 1  | 10^5     | 1.4e-7  | 4.2e-8  |
```
```suggestion
*     | IEEE       | -1, 1  | 10^5     | 2.5e-7  | 5.0e-8  |
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+
+	xx = x;
+	if ( stdlib_base_is_nanf( x ) ) {
+		return 0.0 / 0.0; // NaN
```
```suggestion
		return 0.0f / 0.0f; // NaN
```

Suffixes are important here and below, as otherwise you cause implicit casts. Wherever floats are used and you want to retain the same precision, you need to include suffixes. This also applies below for comparison operators.

gunjjoshi replied:
I have made the changes now, thanks for pointing this out.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,137 @@
+	if ( xx > 0.0 ) {
+		sgn = 1;
+		a = xx;
+	}
+	else {
```
```suggestion
	} else {
```

You can remove linebreaks here and below. We are consistent with JS and C styles here.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/test/fixtures/julia/runner.jl:
```diff
@@ -0,0 +1,73 @@
+gen( x, "data.json" );
+
+# Generate fixture data for small negative values:
+x = range( -1e-200, stop = -1e-208, length = 2003 );
+gen( x, "small_negative.json" );
+
+# Generate fixture data for small negative values:
+x = range( 1e-300, stop = 1e-308, length = 2003 );
```
We need to adjust these tests. e-300 and e-308 are out-of-range for single-precision. All these values underflow to zero. @Planeshifter is right. We should generate single-precision test fixtures, not double-precision.

kgryte replied:
@gunjjoshi You'll want to install Julia in order to run this script after updating it to generate single-precision values. Once installed, you can run this script using

```
$ make test-fixtures-julia TESTS_FIXTURES_FILTER=".*/math/base/special/asinf/.*"
```

LMK if that doesn't work. Recalling from memory.

gunjjoshi replied:
Sure, thanks for sharing @kgryte . Also, we should use the range e-30 to e-38, right ? Along with that, any changes in range required for the range in L68 ?

kgryte replied:
That's right. And let's do `-20` to `-28`.

gunjjoshi replied:
I tried doing this @kgryte. But, getting this error message :

```bash
gunjjoshi@Gunj-Joshis-MacBook stdlib % make test-fixtures-julia TESTS_FIXTURES_FILTER=".*/math/base/special/asinf/.*"

Generating test fixtures: /Users/gunjjoshi/gsoc24/stdlib/lib/node_modules/@stdlib/math/base/special/asinf/test/fixtures/julia/runner.jl
ERROR: LoadError: ArgumentError: Package JSON not found in current path.
- Run `import Pkg; Pkg.add("JSON")` to install the JSON package.
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1772 [inlined]
 [2] macro expansion
   @ ./lock.jl:267 [inlined]
 [3] __require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1753
 [4] #invoke_in_world#3
   @ ./essentials.jl:926 [inlined]
 [5] invoke_in_world
   @ ./essentials.jl:923 [inlined]
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1746
in expression starting at /Users/gunjjoshi/gsoc24/stdlib/lib/node_modules/@stdlib/math/base/special/asinf/test/fixtures/julia/runner.jl:19
make: *** [test-fixtures-julia] Error 1
```

kgryte replied:
Ah, right. You need to do what the error message says. Open `julia` from the terminal and run the `import Pkg` command. That installs the dep.

gunjjoshi replied:
Yes, that worked @kgryte. I have pushed the changes.

kgryte replied:
@gunjjoshi Did you check whether we can adjust the test tolerances?

gunjjoshi replied:
For `data.json`, anything less thatn `3.3` fails. For `small_negative` and `large_negative`, no explicit multiplication is required, so I removed it.

kgryte replied:
Thanks for checking. Did you run code coverage?

```
$ make test-cov TESTS_FILTER=".*/math/base/special/asinf/.*"
$ make view-cov
```

Just want to make sure we have tested all implementation branches.

gunjjoshi replied:
Yes @kgryte, did it now.

kgryte replied:
That is a local file, so I cannot view.

gunjjoshi replied:
Passes all tests, 


88.51% Statements 5165/5835
82.85% Branches 145/175
34.54% Functions 19/55
88.51% Lines 5165/5835

kgryte replied:
You can take a screenshot and upload in this thread, or you can just tell me that we have 100% line and branch coverage. :)

kgryte replied:
Just focus on `math/base/special/asinf`, not the total lines, as that is not what we care about here.

gunjjoshi replied:
![Uploading Screenshot 2024-03-25 at 14.35.30.png…]()

kgryte replied:
Hmm...didn't seem to work.

gunjjoshi replied:
Where can be a possible issue ?

kgryte replied:
Not certain. Probably better if I just pull down and verify. Regardless, I already spotted that we are unlikely to have full branch coverage, and left a comment in the tests.

gunjjoshi replied:
Sure, thanks, I will be adding the tests.

kgryte replied:
@Planeshifter This is a good example of where having a slash command to generate a coverage report would be useful.

kgryte reviewed (changes requested):
Thanks for working on this, @gunjjoshi. Left some initial comments. This is close to the finish line. Needs some clean-up and we need to update the test fixtures.

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/asinf/src/main.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/asinf/src/main.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/asinf/src/main.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/asinf/src/main.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/asinf/src/main.c

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
refactor: added float suffixes

gunjjoshi committed:
refactor: line breaks for if-else

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,135 @@
+	}
+	if ( a < 1.0e-4f ) {
+		z = a;
+		return ( sgn == 1 ) ? z : -z;
```
```suggestion
		return ( sgn == 1 ) ? a : -a;
```

Unnecessary assignment.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,134 @@
+		flag = 1;
+	} else {
+		xx = a;
+		z = xx * xx;
```
```suggestion
		z = a * a;
```

Unnecessary assignment.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,133 @@
+		xx = stdlib_base_sqrtf( z );
+		flag = 1;
+	} else {
+		z = a * a;
```
```suggestion
		xx = a;
		z = a * a;
```

My bad. I see it's used below.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

gunjjoshi committed:
Merge branch 'develop' into asinf-2

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,134 @@
+		a = -xx;
+	}
+	if ( a > 1.0f ) {
+		return 0.0f;
```
I think we should stray from Cephes here. In the double-precision implementation, we return `NaN`. IMO, we should do so here, as well.

gunjjoshi replied:
Sounds good. I'll change it now.

gunjjoshi replied:
Done.

gunjjoshi committed:
Update main.c

Signed-off-by: GUNJ JOSHI <gunjjoshi8372@gmail.com>

gunjjoshi committed:
refactor: changed test fixtures to make them suitable for single precision

gunjjoshi committed:
refactor: ran the polyeval script again, after changes

kgryte committed:
refactor: remove unnecessary variable

Signed-off-by: Athan <kgryte@gmail.com>

gunjjoshi committed:
refactor: adjusted tolerances

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -0,0 +1,131 @@
+		return 0.0f / 0.0f; // NaN
+	}
+	if ( ax < 1.0e-4f ) {
+		return ( sgn == 1 ) ? ax : -ax;
```
```suggestion
		return x;
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/test/test.js:
```diff
@@ -0,0 +1,128 @@
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
```
You'll want to add another test block to verify that the function returns `NaN` when provided a value greater than `1`. And similarly, when provided a value less than `-1`.

gunjjoshi replied:
Aha ! Got it @kgryte :

<img width="1710" alt="Screenshot 2024-03-25 at 14 52 31" src="https://github.com/stdlib-js/stdlib/assets/67432819/d58c7f5d-65d8-4a82-98d3-030c8b295989">

kgryte replied:
Ah, I think I know what the issue is. You need to wait for the screenshot to finish uploading BEFORE you submit your comment. It seems you are submitting your comment before the screenshot has finished uploading.

gunjjoshi replied:
Ohh, sorry for that, uploaded it correctly now.

kgryte replied:
Nice!

gunjjoshi committed:
refactor: added tests for NaN

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js:
```diff
@@ -104,8 +104,8 @@ function asinf( x ) {
 	}
-	if ( a > 1.0 ) {
-		return 0.0;
+	if ( a > 1.0 || a < -1.0 ) {
```
This isn't necessary. `a` is the absolute value.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/src/main.c:
```diff
@@ -108,7 +108,7 @@ float stdlib_base_asinf( const float x ) {
 		ax = -x;
 	}
-	if ( ax > 1.0f ) {
+	if ( ax > 1.0f || ax < -1.0f ) {
```
Same comment as above.

gunjjoshi replied:
Changed it now, thanks !

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/test/test.js:
```diff
@@ -126,3 +127,25 @@ tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		v = -( randu() * 1.0e6 ) - ( 1.0 - EPS );
```
```suggestion
		v = uniform( -1.0e6, -1.0-EPS );
```

Let's clean this up by using `@stdlib/random/base/uniform` here, rather than `randu()`. Same for below.

gunjjoshi replied:
Modified this, along with `test.native.js` too, @kgryte.

gunjjoshi committed:
refactor: absolute values, checking negatives not needed

gunjjoshi committed:
refactor: used uniform instead of randu

kgryte committed:
refactor: remove unnecessary branches and rename variable

This commit ensures the JS implementation stays a bit closer to the C implementation.

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/test/test.js:
```diff
@@ -0,0 +1,151 @@
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		v = uniform( 1.0e6, 1.0 + EPS );
```
```suggestion
		v = uniform( 1.0+EPS, 1.0e6 );
```

`a` must be less than `b`.

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/test/test.native.js:
```diff
@@ -0,0 +1,160 @@
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		v = uniform( 1.0e6, 1.0 + EPS );
```
```suggestion
		v = uniform( 1.0+EPS, 1.0e6 );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM. Thanks, @gunjjoshi! Will merge once CI passes.

gunjjoshi commented:
> LGTM. Thanks, @gunjjoshi! Will merge once CI passes.

Lot's of learnings from this one. Will keep all this in mind for future contributions, thanks for helping out !

kgryte commented:
@gunjjoshi Indeed. Trial by fire.
